docs/commands.md

```diff
@@ -184,3 +184,53 @@ installed version.
   }
 }
 ```
+
+# Client Side Commands
```

Comment:
I didn't realize when we were discussing this on Slack that you meant implementation notes for LS maintainers - apologies! You were right that this isn't the best place as this is primarily for client-side maintainers. 😄 

We don't have any content for server maintainers yet, aside from the [architecture doc](https://github.com/hashicorp/terraform-ls/blob/main/docs/architecture.md). Would you mind copying this into a new one then? You can probably link it from the architecture doc.

We could call it `language-server.md`, to align the name with https://github.com/hashicorp/terraform-ls/blob/main/docs/language-clients.md

Reply:
That said I think that some brief docs for client maintainers would be still helpful here, perhaps 1 section to describe the refresh mechanism in general and/or mention of the relevant capability under each relevant command.